src/js_app.hpp

```diff
@@ -160,18 +171,23 @@ void AppClass<T>::constructor(ContextType ctx, ObjectType this_object, Arguments
         throw std::runtime_error("Expected either a configuration object or an app id string.");
     }
 
-    Protected<typename T::GlobalContext> protected_ctx(Context<T>::get_global_context(ctx));
-    config.transport_generator = [=] {
-        return std::make_unique<NetworkTransport>(protected_ctx);
+    auto realm_constructor = js::Value<T>::validated_to_object(ctx, js::Object<T>::get_global(ctx, "Realm"));
+    
+    Protected<typename T::GlobalContext> protected_ctx(Context::get_global_context(ctx));
+    config.transport_generator = [protected_ctx] {
+        return AppClass<T>::transport_generator(protected_ctx);
     };
 
-    auto realm_constructor = js::Value<T>::validated_to_object(ctx, js::Object<T>::get_global(ctx, "Realm"));
     std::string user_agent_binding_info;
     auto user_agent_function = js::Object<T>::get_property(ctx, realm_constructor, "_createUserAgentDescription");
     if (js::Value<T>::is_function(ctx, user_agent_function)) {
         auto result = js::Function<T>::call(ctx, js::Value<T>::to_function(ctx, user_agent_function), realm_constructor, 0, nullptr);
         user_agent_binding_info = js::Value<T>::validated_to_string(ctx, result);
     }
+    else {
+        // The `_createUserAgentDescription` method is undefined when chrome debugging a React Native app
```

Comment:
this function needs to accommodate that it might get executed in RN environment and we should call it here

Reply:
I don't understand this comment ☝️  Can I get you to rephrase it?

Reply:
In the past, a sync session should be opened without setting `user_agent_binding_info` so we silently skipped the call when the RN debugger was active: https://github.com/realm/realm-js/blob/v6/src/js_sync.hpp#L62-L65